### DISCUSSION TITLE
What is the main difference between source and install

### DISCUSSION THREAD

**Discussion by RayZ0rr** [marlonrichert/zsh-snap#243] [Q&A] [answered]
I am simple zsh user. My main plugins are :

```zsh
zsh-users/zsh-autosuggestions     
zsh-users/zsh-completions         
zsh-users/zsh-syntax-highlighting 
```

These are already complicated because of `compinit` and `source` order importance (at the very end of `.zshrc`) of `zsh-syntax-highlighting `

What command should I use to install these?

[Example zshrc](https://github.com/marlonrichert/zsh-snap/blob/main/.zshrc) has no mention of `install` . however it is mentioned in the [README](https://github.com/marlonrichert/zsh-snap#zshrc-optimization)

**Comment by marlonrichert** [comment, marked as answer]
`source` is for starting plugins. `install` is for making new commands or completions available to you. The latter originally it made symlinks to executables and functions, like a package manager would. However, I decided that is not the best way to do things, since it requires all kinds of maintenance. Thus, now it simply adds things to your `$path` or `$fpath` and autoloads functions for you. As such, it really doesn't contain much logic anymore that you couldn't add to your `.zshrc` file yourself. I will probably deprecate it soon and then remove it some time later.

So, given the repos you listed, this is the correct code to add to your `.zshrc` file:
```sh
znap clone zsh-users/zsh-completions
$fpath=( ~[zsh-users/zsh-completions] $fpath )
    
znap source zsh-users/zsh-autosuggestions     
znap source zsh-users/zsh-syntax-highlighting 
```
